app.py

```diff
@@ -300,24 +377,57 @@ def create_missing_model_ids(self):
 
         return self
 
+    def asset_line_exists(self, record):
+        """
+            Searches the asset catalog for
+            a line item matching the provided `record`.
+
+            Determines if a line is the same if either the
+            serial number or the asset tag was previously
+            recorded, but only if the tag is not empty/generic.
+
+            Returns True if there is an existing record in Odoo,
+            False otherwise.
+        """
+        logging.debug('Checking if "%s" already exists before creation in Odoo', (record.serial))
+        result = self.api.do_search(
+            'erpwarehouse.asset',
+            [
+                ('catalog', '=', ASSET_CATALOG_ID),
+                ('make', '=', self.get_id_from_model(record.model)),
+                '|',
+                ('serial', '=ilike', record.serial),
+                ('tag', '=ilike', record.asset_tag),
+                ('tag', 'not in', [False, '', 'N/A']),
```

Comment:
Remove this line. I think it's partially related to why records aren't getting uploaded